dev/DynamicDependency/DataStore.cpp

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "DataStore.h"
+
+#include "DynamicDependencyDataStore_h.h"
+#include "utf8.h"
+#include "winrt_msixdynamicdepednency.h"
+
+#include <wil/winrt.h>
+
+#include <shlobj.h>
+
+MddCore::PackageDependency MddCore::DataStore::Load(PCWSTR packageDependencyId)
+{
+    std::filesystem::path relativeFilename{ L"DynamicDependency" };
+    relativeFilename /= std::wstring(packageDependencyId) + DataStore::fileExtension;
+
+    auto filename{ GetDataStorePathForUser() };
+    filename /= relativeFilename;
+    wil::unique_hfile file{ OpenFileIfExists(filename.c_str()) };
+    if (!file)
+    {
+        filename = GetDataStorePathForSystem();
+        filename /= relativeFilename;
+        file.reset(OpenFileIfExists(filename.c_str()));
+        if (!file)
+        {
+            // Not found
+            return PackageDependency();
+        }
+    }
+
+    LARGE_INTEGER fileSize{};
+    THROW_IF_WIN32_BOOL_FALSE(::GetFileSizeEx(file.get(), &fileSize));
+    const auto dataSize{ fileSize.QuadPart };
+    THROW_HR_IF(HRESULT_FROM_WIN32(ERROR_INVALID_DATA), dataSize > INT32_MAX);
+    if (dataSize == 0)
+    {
+        // 0-byte file is invalid. Perhaps power was lost when written but before flushed?
+        // 'Fix' it i.e. delete it and report not-found
+        file.reset();
+        std::filesystem::remove(filename);
+        return PackageDependency();
+    }
+
+    const auto bufferSize{ static_cast<DWORD>(dataSize) + 1 };
+    std::unique_ptr<char[]> bufferUtf8{ new char[bufferSize] };
```

Comment:
[Avoid bare new](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#r11-avoid-calling-new-and-delete-explicitly). Something like std::vector would be better.

Reply:
std::make_unique<char>(bufferSize) it is